### PR TITLE
Streamline deployment pipelines

### PR DESCRIPTION
**Pull request by lennym**
* Reuse single image for deployments to all environments
* Add integration test run following dev and test deployments
* Use common failure notification workflow to suport notification of failure at any point in the pipeline

### CODE REVIEW

File: .github/workflows/deploy-dev.yml [hunk 1]
```diff
@@ -120,29 +120,28 @@ jobs:
...
-          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
-          SLACK_TITLE: "dev deployment: failed"
-          SLACK_MESSAGE: "UI: ${{needs.ui.result}}\nCKAN: ${{needs.ckan.result}}"
-          MSG_MINIMAL: "true"
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+          cache: 'npm'
+          cache-dependency-path: ui/package-lock.json
```
Comment (roc) [inline .github/workflows/deploy-dev.yml:136]: Ooo! 💘

File: .github/workflows/notify-fail.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: notify-fail
```
Comment (roc) [inline .github/workflows/notify-fail.yml:1]: Ah this is so cool. I didn't realise you could just put an independent job in ref the other jobs in workflows. Really nice to have it abstracted out to another file.

File: .github/workflows/deploy-dev.yml (hunk 1 above)
Comment (lennym) [reply .github/workflows/deploy-dev.yml:136]: It seemed more sensible than having a one entry matrix.

**Review by roc — APPROVED** [review]
Excellent ymlng!

**Pull request merged**